src/web/WebSettingsService.cpp

```diff
@@ -88,90 +87,109 @@ void WebSettings::read(WebSettings & settings, JsonObject root) {
 
 // call on initialization and also when settings are updated via web or console
 StateUpdateResult WebSettings::update(JsonObject root, WebSettings & settings) {
-    // load the version of the settings
-    // will be picked up in System::check_upgrade()
+    // load the version from the settings config. This can be blank and later used in System::check_upgrade()
     settings.version = root["version"] | EMSESP_DEFAULT_VERSION;
 
-    // load default GPIO configuration based on board profile
-    std::vector<int8_t> data; //  // led, dallas, rx, tx, button, phy_type, eth_power, eth_phy_addr, eth_clock_mode
-    settings.board_profile = root["board_profile"] | EMSESP_DEFAULT_BOARD_PROFILE;
-    // for -D compile setting store it in NVS
-    if ((String)EMSESP_DEFAULT_BOARD_PROFILE != "default" && EMSESP::nvs_.getString("boot") == "") {
-        EMSESP::nvs_.putString("boot", (const char *)EMSESP_DEFAULT_BOARD_PROFILE);
-    }
-
 #ifndef EMSESP_STANDALONE
     bool psram = ESP.getPsramSize() > 0; // System::PSram() is initialized later
 #else
     bool psram = false;
 #endif
 
-    if (System::load_board_profile(data, settings.board_profile.c_str())) {
-        if (settings.board_profile == "CUSTOM") { //read pins, fallback to S32
-            data = {(int8_t)(root["led_gpio"] | 2),
-                    (int8_t)(root["dallas_gpio"] | 18),
-                    (int8_t)(root["rx_gpio"] | 23),
-                    (int8_t)(root["tx_gpio"] | 5),
-                    (int8_t)(root["pbutton_gpio"] | 0),
-                    (int8_t)(root["phy_type"] | PHY_type::PHY_TYPE_NONE),
-                    (int8_t)(root["eth_power"] | 0),
-                    (int8_t)(root["eth_phy_addr"] | 0),
-                    (int8_t)(root["eth_clock_mode"] | 0)};
+#ifdef EMSESP_DEBUG
+    EMSESP::logger().debug("NVS boot value is %s, board profile is %s, EMSESP_DEFAULT_BOARD_PROFILE is %s",
+                           EMSESP::nvs_.getString("boot").c_str(),
+                           root["board_profile"].as<const char *>(),
```

Comment:
This crashes on empty FS, const char* returns a nullptr. Better `root["board_profile"].as<String>().c_str(),`

Reply:
I had it as `.as<String>().c_str()` first but it breaks standalone

Reply:
I'll find another way

Reply:
will this also break: `as<std::string>().c_str()`?

Reply:
fixed.

Reply:
I can't get it to crash - if you could please re-test to make sure I'm not missing anything